resources/js/Components/Detail/CampaignDetail/CampaignDetailBox.tsx

```diff
@@ -13,59 +17,100 @@ const StyledTypography = styled(Typography)({
 
 type CampaignDetailBoxPropType = {
     campaign: CampaignData
+    userCharacter: UserCharacterData
+    setIsEditDrawerOpen: (payload: boolean) => void
 }
 
-const CampaignDetailBox = () => (
-    // {campaign}: CampaignDetailBoxPropType
-    <Box sx={{p: 5, backgroundColor: 'primary'}}>
-        <Grid container columnSpacing={1} rowSpacing={6} xs={6}>
-            <Grid item xs={12}>
-                <StyledTypography>NAME</StyledTypography>
-                <Typography>
-                    {/* {campaign.title} */}
-                    Campaign 1
-                </Typography>
-            </Grid>
-            <Grid item xs={12}>
-                <StyledTypography>SESSIONS PLAYED</StyledTypography>
-                <Typography>
-                    {/* {campaign.sessions_played} */}
-                    25
-                </Typography>
-            </Grid>
-            <Grid item xs={12}>
-                <StyledTypography>CHARACTERS</StyledTypography>
-                <Typography>
-                    Character 1, Character 27, Character 31, Character 45
-                </Typography>
-            </Grid>
-            <Grid item>
-                <Stack spacing={3} direction='row'>
-                    <Button
-                        variant='contained'
-                        startIcon={<CreateIcon fontSize='small' />}>
-                        INVITE
-                    </Button>
-                    <Button
-                        variant='contained'
-                        startIcon={<CreateIcon fontSize='small' />}>
-                        UPDATE
-                    </Button>
-                    <Button
-                        variant='contained'
-                        startIcon={<HistoryEduIcon fontSize='small' />}>
-                        ENTRY
-                    </Button>
-                    <Button
-                        variant='contained'
-                        startIcon={<AutoStoriesIcon fontSize='small' />}>
-                        DM Entry
-                    </Button>
-                </Stack>
-            </Grid>
-        </Grid>
-    </Box>
-)
+const CampaignDetailBox = ({
+    campaign,
+    userCharacter,
+    setIsEditDrawerOpen,
+}: CampaignDetailBoxPropType) => {
+    const [openInviteSnackbar, setOpenInviteSnackbar] = useState(false)
+
+    const handleClick = () => {
+        navigator.clipboard.writeText(campaign.code)
```

Comment:
also this only copies the code, shouldn't we be copying the url to join the specific campaign

you can get the host name using `window.location.origin` then concat whatever needed to go to the join form url

Reply:
blocked by #418